File: VB/WindowsFormsApplication1/MyDockManager/MyDockLayout.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking.Paint
Imports DevExpress.XtraBars.Styles
Imports DevExpress.Utils
Imports DevExpress.XtraBars.Docking
Imports DevExpress.XtraBars.Docking.Helpers

Namespace WindowsFormsApplication1
	Public Class MyDockLayout
		Inherits DockLayout

		Public Sub New(ByVal dock As DockingStyle, ByVal panel As DockPanel)
			MyBase.New(dock, panel)

		End Sub
		Public Sub New(ByVal dock As DockingStyle, ByVal size As Size, ByVal panel As DockPanel)
			MyBase.New(dock, size, panel)

		End Sub

		Protected Overrides Sub DrawCaption(ByVal painter As DockElementsPainter)
			If TypeOf painter Is MyDockElementsSkinPainter Then
				TryCast(painter, MyDockElementsSkinPainter).CurrentPanel = Panel
			End If
			MyBase.DrawCaption(painter)
		End Sub

	End Class
End Namespace

File: CS/WindowsFormsApplication1/MyDockManager/MyDockLayout.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.Utils;
using DevExpress.XtraBars.Docking;
using DevExpress.XtraBars.Docking.Helpers;

namespace WindowsFormsApplication1
{
    public class MyDockLayout : DockLayout
    {

        public MyDockLayout(DockingStyle dock, DockPanel panel)
            : base(dock, panel)
        {

        }
        public MyDockLayout(DockingStyle dock, Size size, DockPanel panel)
            : base(dock, size, panel)
        {

        }

        protected override void DrawCaption(DockElementsPainter painter)
        {
            if (painter is MyDockElementsSkinPainter)
                (painter as MyDockElementsSkinPainter).CurrentPanel = Panel;
            base.DrawCaption(painter);
        }

    }
}
